### PR TITLE
Tok 464/allow foundation to upgrade sc

### PR DESCRIPTION
**Pull request by DinizSa**
## What

- Allow the contracts to be upgraded by the Foundation and Governor

## Testing

- The upgrader account should be able to upgrade all the upgradeable contracts. Unit tests added to cover these cases.

## Refs

- [Jira ticket](https://rsklabs.atlassian.net/jira/software/projects/TOK/boards/267?assignee=712020%3A42dabdd3-0b32-4060-be21-62cfd46a1751&selectedIssue=TOK-464)

### CODE REVIEW

File: src/governance/GovernanceManagerRootstockCollective.sol [hunk 1]
```diff
@@ -53,94 +61,69 @@ contract GovernanceManagerRootstockCollective is UUPSUpgradeable, IGovernanceMan
...
     function validateChanger(address account_) public view {
         if (account_ != _authorizedChanger && account_ != governor) revert NotAuthorizedChanger();
     }
 
-    /**
-     * @notice Validates if an account is authorized as the KYC approver.
-     * @param account_ The address to be validated.
-     * @dev Reverts with `NotKycApprover` if the account is not the KYC approver.
-     */
+    function validateUpgrader(address account_) public view {
```
Comment (nmagariang) [inline src/governance/GovernanceManagerRootstockCollective.sol:117]: Maybe include in the name that it's actually validating that it's can also be a valid Changer?

Comment (DinizSa) [reply src/governance/GovernanceManagerRootstockCollective.sol:117]: `validateUpgrader` ensures that the `account_ ` has permissions to upgrade contracts (governor + validChanger + upgrader), and is not tied to a specific address. 
The idea for the validate functions is to not have a 1:1 relation with a specific address, but to combine the addresses that can perform an action. ie. validateChanger (governor + validChanger), validateUpgrader (governor + changer + upgrader).
But maybe the function name could be improved to make this disassociation between the role and validation method more clear

Comment (nmagariang) [reply src/governance/GovernanceManagerRootstockCollective.sol:117]: >make this disassociation between the role and validation method

Yeah, that would solve it nicely, not sure what a good option could be though, upgrader sums it up quite well 😅

**Review by antomor — APPROVED** [review]
@DinizSa Thank you for the changes. Apart from what @nmagariang mentioned, LGTM

File: src/governance/GovernanceManagerRootstockCollective.sol (hunk 1 above)
Comment (DinizSa) [reply src/governance/GovernanceManagerRootstockCollective.sol]: I renamed to `validateUpgrader` to  `validateUpgradeAuthorization`, wdyt?

File: src/governance/GovernanceManagerRootstockCollective.sol [hunk 2]
```diff
@@ -53,94 +61,69 @@ contract GovernanceManagerRootstockCollective is UUPSUpgradeable, IGovernanceMan
...
     function updateKYCApprover(address kycApprover_) public onlyValidChanger {
         _updateKYCApprover(kycApprover_);
     }
 
-    /**
-     * @notice Validates if an account is authorized as the governor.
-     * @param account_ The address to be validated.
-     * @dev Reverts with `NotGovernor` if the account is not the governor.
-     */
+    function updateUpgrader(address upgrader_) public onlyValidChanger {
```
[the remark this answers is not in the document]
Comment (DinizSa) [reply src/governance/GovernanceManagerRootstockCollective.sol]: after sync with PO, the upgrader role will only be managed by the upgrader role itself. this was implemented

**Review by jmendiola222 — APPROVED** [review]
LGTM

**Pull request merged**